Sources/TestingMacros/ExitTestCapturedValueMacro.swift

```diff
@@ -52,3 +53,26 @@ public struct ExitTestBadCapturedValueMacro: ExpressionMacro, Sendable {
     return #"Swift.fatalError("Unsupported")"#
   }
 }
+
+/// The implementation of the `#__capturedValue()` macro when the type we
+/// inferred for the value was incorrect.
+///
+/// This type is used to implement the `#__capturedValue()` macro. Do not use it
+/// directly.
+public struct ExitTestIncorrectlyCapturedValueMacro: ExpressionMacro, Sendable {
+  public static func expansion(
+    of macro: some FreestandingMacroExpansionSyntax,
+    in context: some MacroExpansionContext
+  ) throws -> ExprSyntax {
+    let arguments = Array(macro.arguments)
+    let expr = arguments[0].expression
+    let nameExpr = arguments[1].expression.cast(StringLiteralExprSyntax.self)
+
+    // Diagnose that the type of 'expr' is invalid.
+    let name = nameExpr.representedLiteralValue ?? expr.trimmedDescription
+    let capture = ClosureCaptureSyntax(name: .identifier(name))
```

Comment:
This fake `ClosureCaptureSyntax` instance is created because we can't pass such a thing as an expression and we need to use a string instead. It's a temporary abstraction until https://github.com/swiftlang/swift-syntax/issues/3085 is resolved.